platform/storage/posix/src/mender-storage.c

```diff
@@ -248,13 +248,23 @@ mender_storage_set_provides(mender_key_value_list_t *provides) {
 }
 
 mender_err_t
-mender_storage_get_provides(mender_key_value_list_t **provides) {
+mender_storage_set_provides(mender_key_value_list_t *provides) {
+    return mender_set_provides(provides, MENDER_STORAGE_NVS_PROVIDES);
+}
+
+mender_err_t
+mender_storage_set_new_provides(mender_key_value_list_t *new_provides) {
+    return mender_set_provides(new_provides, MENDER_STORAGE_NVS_NEW_PROVIDES);
+}
+
+static mender_err_t
+mender_get_provides(mender_key_value_list_t **provides, const char *filename) {
 
     assert(NULL != provides);
 
     char  *provides_str = NULL;
     size_t provides_length;
-    if (MENDER_OK != mender_storage_read_file(MENDER_STORAGE_NVS_PROVIDES, (void **)&provides_str, &provides_length)) {
+    if (MENDER_OK != mender_storage_read_file(filename, (void **)&provides_str, &provides_length)) {
```

Comment:
A bit out of scope of this PR, but I don't see why this code uses the generic `mender_storage_read_file()` when it is a POSIX-specific code that can use the POSIX-specific code to read the file contents.